explorer/nodewatch/nodewatch/RoutesFacade.py

```diff
@@ -68,15 +69,32 @@ def html_nodes(self):
 			'explorer_endpoint': self.explorer_endpoint
 		})
 
-	def json_nodes(self, role, exact_match=False):
-		"""Returns all nodes with matching role."""
+	def json_nodes(self, **kwargs):
+		"""Returns all nodes with condition."""
 
-		def role_filter(descriptor):
-			return role == descriptor.roles if exact_match else role == (role & descriptor.roles)
+		role = kwargs.get('role')
+		exact_match = kwargs.get('exact_match')
 
-		return list(map(
+		limit = kwargs.get('limit')
+		only_ssl = kwargs.get('only_ssl')
+		order = kwargs.get('order')
+
+		def custom_filter(descriptor):
+			if role is not None:
+				role_condition = role == descriptor.roles if exact_match else role == (role & descriptor.roles)
+				if not role_condition:
+					return False
+
+			return descriptor.is_ssl_enabled if only_ssl else True
```

Comment:
Should we be able to do this?
``role_condition and (descriptor.is_ssl_enabled if only_ssl else True)``

Reply:
I did the updates about the `custom_filter` method.
As `role` is always required, which check `role=none` is pointless.